### ISSUE TITLE
Click on Icon does not work

### ISSUE THREAD

**Issue by simasch** [mvysny/karibu-testing#151]
I tried to click on an icon but the click event is not happening:

```java
Icon removeIcon = _get(Icon.class, s -> s.withId("club-contest-discipline-registration-details-remove-2-2"));
_click(removeIcon);
```

What's the correct way to click on an icon?

**Comment by mvysny** [comment]
That is the correct way to click an icon. Let me check please.

**Comment by mvysny** [comment]
The tests pass - if you have added the listeners via `icon.addClickListener()` then `_click()` should fire them properly.

**Comment by simasch** [comment]
In that case I think I see the problem.

I had to register the listener like this:

```java
 removeIcon.getElement()
      .addEventListener("click", e -> removeListener.accept(disciplineRegistration))
      .addEventData("event.stopPropagation()");
```

Because I must stop the event propagation as the icon is in the summary of the details component.
This issue is still open: https://github.com/vaadin/flow/issues/1363

In that case there is no way that my click listener will receive the even?

**Comment by mvysny** [comment]
You're right, good catch! That's a DOM event listener which needs to be fired a bit differently: https://github.com/mvysny/karibu-testing/tree/master/karibu-testing-v10#firing-dom-events

However, I agree that the current behavior is confusing. Either:

1. `_click()` fires the DOM event listener as well, along with the higher-level `.addClickListener()`, OR
2. `_click()` should contain a huge warning that it doesn't fire the DOM `click` event.

I'm leaning towards the 1. but we need to test this thoroughly, to prevent double-firing of the higher-level click listener. What do you think @simasch  ?

**Comment by simasch** [comment]
1. would be great!

**Comment by mvysny** [comment]
Implemented :+1:  Meanwhile you can fire the DOM click event manually, via https://github.com/mvysny/karibu-testing/tree/master/karibu-testing-v10#firing-dom-events

**Comment by mvysny** [comment]
Implemented in 9c64e80411362d65e54a12ac2d00b39dfb158ba7 , KT 2.0.3/1.3.25

**Comment by simasch** [comment]
Great. Thanks @mvysny
